sympy/combinatorics/rewritingsystem_fsm.py

```diff
@@ -27,39 +28,41 @@ def __init__(self, name, is_start=False, is_dead=False, is_accept=False, subst=N
             self.state_type = "dead"
 
     def add_transition(self, letter, state):
-        """
-        This method is to add a transition from the current state to a new state.
+        '''
+        Add a transition from the current state to a new state.
 
-        Arguments:
-            letter: The alphabet element the current state reads to make the state transition.
-            state: This will be an instance of the State object which represents a new state after in the transition after the alphabet is read.
-        """
+        Keyword Arguments:
+            letter -- The alphabet element the current state reads to make the state transition.
+            state -- This will be an instance of the State object which represents a new state after in the transition after the alphabet is read.
+
+        '''
         self.transitions[letter] = state
 
 class StateMachine(object):
-    """
-    The state machine class which manages the states and their transisitons of the automata.
+    '''
+    Represents the attributes and methods of a finite state machine.
 
     Attributes:
-        states (dictionary): Collection of all registered.
-        name (str): Name of the state machine.
-    """
+        states (dictionary) -- Collection of all registered `State` objects.
+        name (str) -- Name of the state machine.
+    '''
 
     def __init__(self, name):
         self.name = name
         self.states = {} # Contains all the states in the machine.
 
     def add_state(self, state_name, is_start=False, is_dead=False, is_accept=False, subst=None):
-        """
-        This instantiates a state object and stores this in the 'states' list.
+        '''
+        Instantiate a state object and stores it in the 'states' dictionary.
 
         Arguments:
-            state_name (instance of FreeGroupElement or string): name of the new states.
-            is_start (boolean): To mark if the state is a start state.
-            is_dead (boolean): To mark if the state is a dead state.
-            is_accept (boolean): To mark if the state is an accept state.
-            subt (instance of FreeGroupElement): right hand rule for dead state.
-        """
+            state_name (instance of FreeGroupElement or string) -- name of the new states.
+            is_start (boolean) -- To mark if the state is a start state.
+            is_dead (boolean) -- To mark if the state is a dead state.
+            is_accept (boolean) -- To mark if the state is an accept state.
+            subt (instance of FreeGroupElement) -- right hand rule for dead state.
```

Comment:
You changed `subst` to `rh_rule` in the `State` class, so then this should change as well for consistency. Or you could just keep both as `subst` - either way is fine as long aas it's the same.

Also, here's a thought: instead of having 3 different keyword arguments `is_start`, `is_dead` and `is_accept`, you could just have one argument `type` that's set to `None` or `'n'` or `'r'` by default (for "normal" or "regular") and can be set to `start`, `dead` or `accept` (or alternatively `'s',`'d'` and `'a'` - probably worth allowing both the long and short options). What do you think?